server/trucks.js

```diff
@@ -247,10 +248,14 @@ var foodTrucks = [
 		schedule: []
 	}
 ];
+function filterByDay(day) {
+	var filtered = _.filter(foodTrucks, function(data) {
+		var toReturn = _.filter(data.schedule, function(date) {
```

Comment:
Took me a moment to think this through (using the nested filter) but sure enough it works.